src/frontend/src/pages/AdminToolsPage/AdminToolsUserManagement.tsx

```diff
@@ -0,0 +1,142 @@
+import Autocomplete from '@mui/material/Autocomplete';
+import { NERButton } from '../../components/NERButton';
+import { Grid, InputAdornment, Typography } from '@mui/material';
+import PageBlock from '../../layouts/PageBlock';
+import TextField from '@mui/material/TextField';
+import SearchIcon from '@mui/icons-material/Search';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import { useState } from 'react';
+import { useAllUsers, useUpdateUserRole } from '../../hooks/users.hooks';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+import { fullNamePipe } from '../../utils/pipes';
+
+interface UserData {
+  userId: number;
+  role: string;
+}
+
+const AdminToolsUserMangaement: React.FC = () => {
+  const [role, setRole] = useState('');
+  const [user, setUser] = useState<UserData | null>(null);
+  const [isDisabled, setIsDisabled] = useState(true);
+  const [hideSuccessLabel, setHideSuccessLabel] = useState(true);
+  const { isLoading, isError, error, data } = useAllUsers();
+  const update = useUpdateUserRole();
+
+  if (isLoading || !data) return <LoadingIndicator />;
+
+  if (isError) return <ErrorPage message={error?.message} />;
+
+  const handleSearchChange = (event: React.SyntheticEvent<Element, Event>, value: string | null) => {
+    if (value) {
+      const user = data.find((user) => fullNamePipe(user) === value.split(' -')[0]);
+      if (user) setUser(user);
+    } else {
+      setUser(null);
+    }
+  };
+
+  const handleRoleChange = (event: SelectChangeEvent) => {
+    setRole(event.target.value);
+    if (user && event.target.value === user.role) {
+      setIsDisabled(true);
+    } else {
+      setIsDisabled(false);
+    }
+  };
+
+  const handleClick = async () => {
+    setHideSuccessLabel(true);
+    await update.mutateAsync({ userId: user?.userId, role }).catch((error) => {
+      alert(error);
+      throw new Error(error);
+    });
+    setHideSuccessLabel(false);
+  };
+
+  return (
+    <PageBlock title={'Role Management'}>
+      <Grid container spacing={2}>
+        <Grid item xs={12} md={8}>
+          <Autocomplete
+            disablePortal
+            id="autocomplete"
+            onChange={handleSearchChange}
+            options={data.map((user) => `${fullNamePipe(user)} - ${user.email}`)}
+            sx={{
+              height: '40px',
+              backgroundColor: 'ActiveBorder',
+              width: '100%',
+              borderRadius: '25px',
+              borderColor: 'black',
+              '.MuiOutlinedInput-notchedOutline': {
+                border: 0,
+                borderRadius: '25px'
+              }
```

Comment:
additionally for areas like this where the css is getting long and very specific, I wanna start moving them up into object styles so that the code is easier to read. Just move this object with all the css into a variable called autoCompleteStyle, and then call it here